specs/PushNotifications/PushNotifications-spec.md

```diff
@@ -0,0 +1,438 @@
+        {
+            // Initialize using a manifest defined activatorId to Activate the InProc COM server
+            InProcActivatorDetails(String taskClsid);
+
+            // Initialize an inproc activator for LOW-IL apps like UWP with no COM Server specified in manifest
+            InProcActivatorDetails();
+
+            // The CLSID associated with the Client COM server that reunion will adopt
+            String TaskClsid{ get; };
+
+            // The conditions under which Push Triggers would execute
+            Windows.ApplicationModel.Background.IBackgroundCondition Condition;
+
+            // The in-memory Event handler to subscribe to if the app is in the foreground
+            event Windows.Foundation.EventHandler<PushNotificationActivatedEventArgs> PushActivated;
+
+            // The backgroundRegistration that gets populated on successful registration
+            Windows.ApplicationModel.Background.IBackgroundTaskRegistration Registration;
+        };
+
+        enum ChannelStatus
+        {
+            InProgress, // The request is in progress and there is no retry operation
+            InProgressRetry, // The request is in progress and is in a backoff retry state
+            CompletedSuccess, // The request completed successfully
+            CompletedFailure, // The request failed with some critical internal error. Check ExtendedError for HRESULT
+        };
+
+        runtimeclass PushNotificationChannelResult
+        {
+            PushNotificationChannelResult(
+                Windows.Networking.PushNotifications.PushNotificationChannel channel,
+                HRESULT extendedError,
+                ChannelStatus status);
+
+            // The Push channel associated with the Result. Null if InProgress or completion failed
+            Windows.Networking.PushNotifications.PushNotificationChannel Channel { get; };
+
+            // Low-level information about why the operation might not
+            // have worked. Apps generally do not take action based on
+            // this value, in part because the value may not be stable.
+            HRESULT ExtendedError{ get; };
+
+            // The Status of the ChannelComplete operation
+            ChannelStatus Status { get; };
+        };
+
+        static runtimeclass PushNotificationChannelManager
+        {
+            // Register an activator using a Details context
+            static void RegisterPushNotificationActivator(InProcActivatorDetails details);
+
+            // Unregister any activator if present
+            static void UnregisterPushNotificationActivator(Windows.ApplicationModel.Background.IBackgroundTaskRegistration registration);
+
+            // Request a Push Channel with an encoded RemoteId from WNS. RemoteId is an AAD identifier GUID
+            static Windows.Foundation.IAsyncOperationWithProgress<PushNotificationChannelResult, PushNotificationChannelResult> CreatePushNotificationChannelAsync(String remoteId);
```

Comment:
This shouldn't compile - `IAsyncOperationWithProgress` requires a value-type for its "progress" type (the second type parameter.)  So you could define `PushNotificationChannelCreationProgress` as an enum or struct if you'd like.

Reply:
(IIRC we also allow `Object` there to indicate that there's no unique value in the progress information and that the Result has all the pending progress.)